### DISCUSSION TITLE
Problem with adjusting zrange for legend in contour plot.

### DISCUSSION THREAD

**Discussion by Adamszk** [kookma/ogpf#51] [Q&A]
I am trying to replicate matlab code with fortran.  Matlab is using contourf feature which I can translate with ogpf library using contour funtion with fortran.   Numerical data for (x,y,z) used for plot are the same, however the plots looks different.  I realized that when color box are displayed, the range for z data or range of color in legend is different.  For example, all z values are close to 1, matlab z range is from 0 to 2, while Fortran z range in color box is 0.99 to 1.01  So the obvious step is to change the color/z data range in Fortran.  However, I do not see options listed/available to do that.
I tried the following,

`call gp%axis([-1.0,+1.0,-2.0,+2.0,0,+2.0]) ! Sets all axis at same time`   

It does not work (for z data).   I also tried the following,

`call gp%xlim([-1.0,+1.0]) ! Sets only the xrange
call gp%ylim([-2.0,+3.0]) ! Sets only the yrange
call gp%zlim([0,+2.0]) ! Sets only the zrange`

All these calls do not work.

Any suggestions?

Below is the test code for GNU Octave/Matlab

` figure 1;
 %
 % ------------------------------------------------------------------
 % Use this example to test my problem with contourf plotting of data
 % ------------------------------------------------------------------
 %
 %
 %
 %
 % Create a vector of x values
x = linspace(-2*pi,2*pi,2);
% Create a vector of y values 
y = linspace(0,4*pi,2);
% Make matrices with all combinations of x and y values for plotting
[X,Y] = meshgrid(x,y);
Z = sin(X)+cos(Y);

X
Y
Z

colormap ("default");
contourf(X,Y,Z)
title ({"Example 4: contourf() Z = sin(x) + cos(y) "});
colorbar ();
colorbar on;
`
This is a code in Fortran,

`subroutine problem111()
        type(gpf):: gp

        real(wp), allocatable:: x(:,:)
        real(wp), allocatable:: y(:,:)
        real(wp), allocatable:: z(:,:)
        integer:: m
        integer:: n
        real(wp), parameter :: pi=4.0_wp*atan(1.0_wp)
        real(wp), parameter :: ppi=8.0_wp*atan(1.0_wp)
        real(wp), parameter :: ypi=16.0_wp*atan(1.0_wp)
        real(wp), parameter :: zpi=2.0_wp*atan(1.0_wp)
        
        ! create the xyz data
        call meshgrid(x, y, linspace(-ppi,ppi, 2), linspace(0.0_wp,ypi, 2)  )

        m=size(x,1)
        n=size(x,2)
        allocate( z(m,n) )

        z = sin(x)+cos(y)

        call gp%options('unset key')
        call gp%options('unset surface')
        call gp%options('set colorbox')
        call gp%axis([real(wp):: -ppi, ppi, 0, ypi, 0, zpi])
        call gp%title('my contour problem 111 test plot 2')
        call gp%contour(x,y,z, lspec='t "Matlab Jet"', palette='jet')

       ! plot the contour
       ! call gp%options('unset border; unset tics')
       ! call gp%title('My contour problem 111 test plot 1')
       ! call gp%surf(x,y,z, palette='accent')
       
   print*, " rank of x  = ", rank(x)  
   print*, " rank of y  = ", rank(y)  
   print*, " rank of z  = ", rank(z)  
   
   
   print*, " x  = ", x 
   print*, " y  = ", y 
   print*, " z  = ", z  
    end subroutine problem111`

**Comment by kookma** [comment]
You may know ogp actually calles gnuplot! So, would you please give a try directly in gnuplot!
If it works, then we can revise the code here to address this issue.

**Comment by Adamszk** [comment]
The source of my problem is the option that is available in matlab contour which is vn in :

``contourf (x, y, z, vn)"

"The optional input vn is either a scalar denoting the number of contour lines to compute or a vector containing the Z values where lines will be computed.  When vn is a vector the number of contour lines is numel (vn). However, to compute a single contour line at a given value use vn = [val, val]. If vn is omitted it defaults to 10."

I do not see that option vn  available in ogpf function.
Currently I am looking in GNU Plot if that option is there.

**Comment by kookma** [comment]
Yes, please first check the http://gnuplot.info/
If you have the option there, then you can pass it using ogpf to gnuplot.

Would you please check these pages:

* https://gnuplot.sourceforge.net/demo_6.0/contours.html
* http://gnuplot.info/docs_6.0/loc10726.html


A good example is here: https://stackoverflow.com/a/57090231/439976
